numba/tests/test_np_functions.py

```diff
@@ -1590,6 +1602,231 @@ def test_roll_exceptions(self):
             msg = "shift must be an integer"
             assert msg in str(e.exception)
 
+    def test_np_trapz_basic(self):
+        pyfunc = np_trapz
+        cfunc = jit(nopython=True)(pyfunc)
+        _check = partial(self._check_output, pyfunc, cfunc)
+
+        y = [1, 2, 3]
+        _check({'y': y})
+
+        y = (3, 1, 2, 2, 2)
+        _check({'y': y})
+
+        y = np.arange(15).reshape(3, 5)
+        _check({'y': y})
+
+        y = np.linspace(-10, 10, 60).reshape(4, 3, 5)
+        _check({'y': y}, abs_tol=1e-13)
+
+        self.rnd.shuffle(y)
+        _check({'y': y}, abs_tol=1e-13)
+
+        y = np.array([])
+        _check({'y': y})
+
+        y = np.array([3.142, np.nan, np.inf, -np.inf, 5])
+        _check({'y': y})
+
+        y = np.arange(20) + np.linspace(0, 10, 20) * 1j
+        _check({'y': y})
+
+        y = np.array([], dtype=np.complex128)
+        _check({'y': y})
+
+        y = (True, False, True)
+        _check({'y': y})
+
+    def test_np_trapz_x_basic(self):
+        pyfunc = np_trapz_x
+        cfunc = jit(nopython=True)(pyfunc)
+        _check = partial(self._check_output, pyfunc, cfunc)
+
+        y = [1, 2, 3]
+        x = [4, 6, 8]
+        _check({'y': y, 'x': x})
+
+        y = [1, 2, 3, 4, 5]
+        x = (4, 6)
+        _check({'y': y, 'x': x})
+
+        y = (1, 2, 3, 4, 5)
+        x = [4, 5, 6, 7, 8]
+        _check({'y': y, 'x': x})
+
+        y = np.array([1, 2, 3, 4, 5])
+        x = [4, 4]
+        _check({'y': y, 'x': x})
+
+        y = np.array([])
+        x = np.array([2, 3])
+        _check({'y': y, 'x': x})
+
+        y = (1, 2, 3, 4, 5)
+        x = None
+        _check({'y': y, 'x': x})
+
+        y = np.arange(20).reshape(5, 4)
+        x = np.array([4, 5])
+        _check({'y': y, 'x': x})
+
+        y = np.arange(20).reshape(5, 4)
+        x = np.array([4, 5, 6, 7])
+        _check({'y': y, 'x': x})
+
+        y = np.arange(60).reshape(5, 4, 3)
+        x = np.array([4, 5])
+        _check({'y': y, 'x': x})
+
+        y = np.arange(60).reshape(5, 4, 3)
+        x = np.array([4, 5, 7])
+        _check({'y': y, 'x': x})
+
+        y = np.arange(60).reshape(5, 4, 3)
+        self.rnd.shuffle(y)
+        x = y + 1.1
+        self.rnd.shuffle(x)
+        _check({'y': y, 'x': x})
+
+        y = np.arange(20)
+        x = y + np.linspace(0, 10, 20) * 1j
+        _check({'y': y, 'x': x})
+
+        y = np.array([1, 2, 3])
+        x = np.array([1 + 1j, 1 + 2j])
+        _check({'y': y, 'x': x})
+
+    @unittest.skip('NumPy behaviour questionable')
+    def test_trapz_numpy_questionable(self):
+        pyfunc = np_trapz
+        cfunc = jit(nopython=True)(pyfunc)
+        _check = partial(self._check_output, pyfunc, cfunc)
+
+        # passes (NumPy and Numba return 2.0)
+        y = np.array([True, False, True, True]).astype(np.int)
+        _check({'y': y})
+
+        # fails (NumPy returns 1.5; Numba returns 2.0)
```

Comment:
Perhaps report this upstream to NumPy/see if it's on their issue tracker? Having just drawn a picture, I'm pretty sure the answer should be 2.0.